tests/test_installed_distributions.py

```diff
@@ -17,8 +19,13 @@ def test_installed_distributions(pip, some_distribution):
 
     assert distribution.name == some_distribution.name
     assert distribution.version == some_distribution.version
-    assert distribution.location == some_distribution.location
-    assert distribution.editable == some_distribution.editable
+    if distribution.location is not None:
+        # We don't know exactly where the distribution has been installed,
+        # but we know it exists and therefore is editable.
+        assert os.path.exists(distribution.location)
+        assert distribution.editable
+    else:
+        assert not distribution.editable
```

Comment:
Shouldn't all distributions have locations? This probably means we need to make the fixture more realistic instead.

Or does this mean that the behavior here is different for different versions of `pip`?

Reply:
All installed distributions *should* have locations, but yeah: whether or not we can detect them seems to be dependent on the version of `pip`.

In particular:
* Versions below `9.0.0` use the old `list` parser, which doesn't include location information
* Versions from `9.0.0` to `< 10.0.0b1` use the JSON `list` loader, but that output doesn't include the `location` key
* Versions `10.0.0b1` and newer use the same JSON loader, but **do** include the `location` key

So yeah, I think the test should be updated to explicitly test for the expected behavior based on the version of `pip` under test. I can update it to do that.

Reply:
(AFAICT, this is a historical limitation of `pip list` itself on older `pip`s. We might have more luck by additionally wrapping `pip show`, but I'm not sure whether we'd like to do that for the distribution listing API. Thoughts?)

Reply:
I don't think wrapping `pip show` as well is necessary, the main goal of this project is to provide the same functionality as internal pip APIs, adding the locations here is just bonus (also those versions are sufficiently old that I'm OK with locations just missing for them, although we should detail that in the docs.)